src/Common/tests/System/Resources/Binary/TestData.cs

```diff
@@ -0,0 +1,165 @@
+            };
+
+        public static IReadOnlyDictionary<string, object> ByteArrayConverterDrawing
+        {
+            get => new Dictionary<string, object>(ByteArrayConverterWithoutDrawing)
+            {
+                ["bitmap_bytes"] = new Bitmap(Path.Combine("bitmaps", "almogaver24bits.bmp")),
+                ["icon_bytes"] = new Icon(Path.Combine("bitmaps", "32x32_one_entry_4bit.ico"))
+            };
+        }
+
+        public static IReadOnlyDictionary<string, object> StringConverter
+        {
+            // ImageFormatConverter is part of System.Windows.Extensions.
+            get => PlatformDetection.IsDrawingSupported && PlatformDetection.IsWindows ?
+                StringConverterDrawing : StringConverterWithoutDrawing;
+        }
+
+        public static Dictionary<string, object> StringConverterWithoutDrawing { get; } =
+            new Dictionary<string, object>()
+            {
+                ["color_string"] = Color.AliceBlue,
+                ["point_string"] = new Point(2, 6),
+                ["rect_string"] = new Rectangle(3, 6, 10, 20),
+                ["size_string"] = new Size(4, 8),
+                ["sizeF_string"] = new SizeF(4.2f, 8.5f),
+                ["cultureInfo_string"] = new CultureInfo("en-US"),
+                ["enum_string"] = DayOfWeek.Friday
+            };
+
+        public static IReadOnlyDictionary<string, object> StringConverterDrawing
+        {
+            get => new Dictionary<string, object>(StringConverterWithoutDrawing)
+            {
+                ["imageFormat_string"] = ImageFormat.Png,
+                ["font_string"] = SystemFonts.DefaultFont
+            };
+        }
+
+        public static IReadOnlyDictionary<string, (Type type, Stream stream)> Stream
+        {
+            get => PlatformDetection.IsDrawingSupported ?
+                StreamDrawing : StreamWithoutDrawing;
+        }
+
+        public static Dictionary<string, (Type type, Stream stream)> StreamWithoutDrawing { get; } =
+            new Dictionary<string, (Type type, Stream stream)>()
+            {
+                ["myResourceType_stream"] = (typeof(MyResourceType), new MemoryStream(new byte[] { 0xAB, 0xCD, 0xEF, 0x01, 0x23, 0x45, 0x67, 0x89 }))
+            };
+
+        public static IReadOnlyDictionary<string, (Type type, Stream stream)> StreamDrawing
+        {
+            get => new Dictionary<string, (Type type, Stream stream)>(StreamWithoutDrawing)
+            {
+                ["icon_stream"] = (typeof(Icon), File.OpenRead(Path.Combine("bitmaps", "32x32_one_entry_4bit.ico"))),
+                ["bitmap_stream"] = (typeof(Bitmap), File.OpenRead(Path.Combine("bitmaps", "almogaver24bits.bmp")))
+            };
+        }
+        
+        public static void WriteResources(string file)
+        {
+            using (var writer = new BinaryResourceWriter(file))
+            {
```

Comment:
All these calls to GetType().AssemblyQualifiedName need to check TypeForwardedFrom.  That's what's causing the test failures on desktop.